Civi/Install/Requirements.php

```diff
@@ -227,8 +227,7 @@ public function checkMysqlConnection(array $db_config) {
       'details' => "Connected",
     );
 
-    $host = isset($db_config['server']) ? $db_config['server'] : $db_config['host'];
-    $conn = @mysqli_connect($host, $db_config['username'], $db_config['password']);
+    $conn = @mysqli_connect($host, $db_config['username'], $db_config['password'], $db_config['database'], $db_config['port']);
```

Comment:
How can it use `$host` if `$host` is no longer set? (Guess: On some deployments which use localhost or have some `php.ini` settings, it doesn't matter? But generally it should?)

Thought: since this code is repeated a lot and has been the subject of multiple revisions, perhaps it should move to a common function?

```php
  $conn = $this->connect($db_config);

  ...

  protected function connect($db_config) {
     $host = isset($db_config['server']) ? $db_config['server'] : $db_config['host']; 
     $conn = @mysqli_connect($host, $db_config['username'], $db_config['password'], $db_config['database'], $db_config['port']);
     return $conn;
  }